src/literal.h

```diff
@@ -693,6 +693,12 @@ class Literals : public SmallVector<Literal, 1> {
   Literals(size_t initialSize) : SmallVector(initialSize) {}
 
   Type getType() {
+    if (empty()) {
+      return Type::none;
+    }
+    if (size() == 1) {
+      return (*this)[0].type;
+    }
     std::vector<Type> types;
```

Comment:
Would it be better to make it `SmallVector` too?

Reply:
The class itself here is a SmallVector. The problem is the code it calls only accepts a normal one, so it converts, which is kind of annoying... perhaps we could add a template to support more inputs there, or something like that. Could be worth looking into.

(Regardless this change remains useful I think since it avoids the Type() constructor entirely in common cases.)